libraries/lib-effects/MixAndRender.cpp

```diff
@@ -95,34 +95,15 @@ TrackListHolder MixAndRender(const TrackIterRange<const WaveTrack> &trackRange,
       oneinput = true;
    // only one input track (either 1 mono or one linked stereo pair)
 
-   // EmptyCopy carries over any interesting channel group information
-   // Maybe first is stereo, or it is mono, but some other input is stereo;
-   // therefore don't use WideEmptyCopy, but make and link a right channel as
-   // needed
-   // And reset pan and gain
-   auto mixLeft =
-      first->EmptyCopy(trackFactory->GetSampleBlockFactory(), false);
-
-   // TODO: more-than-two-channels
-
-   decltype(mixLeft) mixRight{};
-   if (!mono)
-      mixRight = trackFactory->Create(format, rate);
-
-   auto result = TrackList::Temporary(nullptr, mixLeft, mixRight);
-
-   mixLeft->SetPan(0);
-   mixLeft->SetGain(1);
-   mixLeft->SetRate(rate);
-   mixLeft->ConvertToSampleFormat(format);
-   if (oneinput)
-      mixLeft->SetName(first->GetName()); /* set name of output track to be the same as the sole input track */
-   else
-      /* i18n-hint: noun, means a track, made by mixing other tracks */
-      mixLeft->SetName(newTrackName);
-   mixLeft->MoveTo(mixStartTime);
-
-   auto maxBlockLen = mixLeft->GetIdealBlockSize();
+   auto result = trackFactory->Create(mono ? 1 : 2, *first);
```

Comment:
Now you are assuming that `format` and `rate`, which were formerly passed to `Create`, are the same as the format and rate of `first`.  But is this so?  Should we examine all uses of `MixAndRender`?

I was expecting instead that the new overload would take format and rate, and use the prototype only to copy other things as in `WaveTrack::Init`.